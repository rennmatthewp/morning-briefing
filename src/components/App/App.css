.app {
  background-color: #D4D4D4;
  margin: 0 auto;
  padding: 5px 20px;
}

header {
  width: 90vw;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
}

h1 {
  font-size: 36px;
  line-height: 1.2;
  width: 145px;
}