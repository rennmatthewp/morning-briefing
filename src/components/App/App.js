import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { populateWeather } from '../../actions';
import { getWeatherData, cleanWeatherData } from '../../helper/apiCalls';
import NewsContainer from '../../containers/NewsContainer/NewsContainer';
//eslint-disable-next-line
import WeatherContainer from '../../containers/WeatherContainer/WeatherContainer';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      errorStatus: null
    };
  }

  componentDidMount() {
    this.getWeather();
  }

  getWeather = async () => {
    try {
      const weatherResponse = await getWeatherData();
      const cleanedWeatherData = await cleanWeatherData(weatherResponse);
      this.props.populateWeather(cleanedWeatherData);
    } catch (error) {
      this.setState({ errorStatus: error });
    }
  };

  render() {
    return (
      <div className="app">
        <header>
          <h1>Morning Briefing</h1>
          <Route path="/" component={WeatherContainer} />
        </header>
        <Route path="/" component={NewsContainer} />
      </div>
    );
  }
}

App.propTypes = {
  populateWeather: PropTypes.func
};

export const mapDispatchToProps = dispatch => ({
  populateWeather: weather => dispatch(populateWeather(weather))
});

export default withRouter(connect(null, mapDispatchToProps)(App));
