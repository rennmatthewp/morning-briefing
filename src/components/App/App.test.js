import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let mockPopulateNews;
  let mockPopulateWeather;
  let renderedApp;

  beforeEach(() => {
    mockPopulateNews = jest.fn();
    mockPopulateWeather = jest.fn();
    return (renderedApp = shallow(
      <App
        populateNews={mockPopulateNews}
        populateWeather={mockPopulateWeather}
      />
    ));
  });

  it('should match the snapshot', () => {
    expect(renderedApp).toMatchSnapshot();
  });
});
