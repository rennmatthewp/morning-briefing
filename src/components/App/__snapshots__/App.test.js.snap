// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    populateNews={[Function]}
    populateWeather={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                    Morning Briefing
          </h1>
          <Route
                    component={[Function]}
                    path="/"
          />
</header>,
        <Route
          component={[Function]}
          path="/"
/>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Morning Briefing
</h1>,
            <Route
              component={[Function]}
              path="/"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Morning Briefing",
            },
            "ref": null,
            "rendered": "Morning Briefing",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                        Morning Briefing
            </h1>
            <Route
                        component={[Function]}
                        path="/"
            />
</header>,
          <Route
            component={[Function]}
            path="/"
/>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Morning Briefing
</h1>,
              <Route
                component={[Function]}
                path="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Morning Briefing",
              },
              "ref": null,
              "rendered": "Morning Briefing",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
